pandas/_libs/tslib.pyx

```diff
@@ -3436,7 +3436,24 @@ def cast_to_nanoseconds(ndarray arr):
     return result
 
 
-def pydt_to_i8(object pydt):
+cdef inline _to_i8(object val):
+    cdef pandas_datetimestruct dts
+    try:
+        return val.value
+    except AttributeError:
+        if is_datetime64_object(val):
```

Comment:
just ``Timestamp(val).value`` is prob enough here, this looks like really old conversion code

Reply:
did u fix for this?

Reply:
See https://github.com/pandas-dev/pandas/pull/17711#issuecomment-333028188

Reply:
so just change this to (to remove all of this code)

```
try
    return val.value
except AttributeError:
      if not is_string_object(val):
         return Timestamp(val).value
   return val
```

Reply:
Implemented this and it doesn't appear to have broken anything.

Reply:
great, ok merging this. would still like to remove this routing (in favor of pydt_to_i8 but can do that in another PR.